Allow accelerator to instantiate the device

nelyahu opened the pull request:
when instantiating torch.device for HPU it cannot be fed with HPU:1 annotation, but only "HPU".
moving the logic to accelerator will allow to solve this issue, with single line change.

loadams commented on deepspeed/runtime/engine.py:
```diff
@@ -977,13 +977,13 @@ def _set_distributed_vars(self, args):
         else:
             self.world_size = 1
             self.global_rank = 0
-            self.device = torch.device(get_accelerator().device_name())
```
FYI @nelyahu - the CI is working again, the error seems to be caused by these changes, can you take a look, please?

loadams replied:
Thanks!

loadams commented:
Will merge #5275 first, that way we can have the newly added unit tests run on this, then we can merge it?

nelyahu commented:
@loadams Sure, thanks :)

loadams commented:
> @loadams Sure, thanks :)

Done now, so the tests should run on this PR, then we can merge it, thanks!

nelyahu commented:
Halting this PR till habana version 1.16.0 will be released.

loadams commented:
> Halting this PR till habana version 1.16.0 will be released. CI issue should be fixed there.

Thanks @nelyahu - when it is released, go ahead and just update it in this PR?  Also curious if you can share what is required in that release that is causing failures here since it appears the tests just hang now?

nelyahu commented:
> > Halting this PR till habana version 1.16.0 will be released. CI issue should be fixed there.
> 
> Thanks @nelyahu - when it is released, go ahead and just update it in this PR? Also curious if you can share what is required in that release that is causing failures here since it appears the tests just hang now?
> 
> And just tag one of us when it is updated and we can get this merged.

Sure @loadams , i will update.
I verified this patch on v1.16.0, as our dev process is on the next release branch, and i did not verified v1.14.0 (the release which is being tested here). There was a lot of bug fixed since then, cannot say for sure what is the root cause.

loadams commented:
> > > Halting this PR till habana version 1.16.0 will be released. CI issue should be fixed there.
> > 
> > 
> > Thanks @nelyahu - when it is released, go ahead and just update it in this PR? Also curious if you can share what is required in that release that is causing failures here since it appears the tests just hang now?
> > And just tag one of us when it is updated and we can get this merged.
> 
> Sure @loadams , i will update. I verified this patch on v1.16.0, as our dev process is on the next release branch, and i did not verified v1.14.0 (the release which is being tested here). There was a lot of bug fixed since then, cannot say for sure what is the root cause.

Thanks @nelyahu - feel free to tag us when it the new version is released and updated so we can merge this then, thanks!

loadams commented:
> > > > Halting this PR till habana version 1.16.0 will be released. CI issue should be fixed there.
> > > 
> > > 
> > > Thanks @nelyahu - when it is released, go ahead and just update it in this PR? Also curious if you can share what is required in that release that is causing failures here since it appears the tests just hang now?
> > > And just tag one of us when it is updated and we can get this merged.
> > 
> > 
> > Sure @loadams , i will update. I verified this patch on v1.16.0, as our dev process is on the next release branch, and i did not verified v1.14.0 (the release which is being tested here). There was a lot of bug fixed since then, cannot say for sure what is the root cause.
> 
> Thanks @nelyahu - feel free to tag us when it the new version is released and updated so we can merge this then, thanks!

@nelyahu - I tried 1.15.1, looks like same errors, so we will wait for 1.16

nelyahu commented:
> > > > > Halting this PR till habana version 1.16.0 will be released. CI issue should be fixed there.
> > > > 
> > > > 
> > > > Thanks @nelyahu - when it is released, go ahead and just update it in this PR? Also curious if you can share what is required in that release that is causing failures here since it appears the tests just hang now?
> > > > And just tag one of us when it is updated and we can get this merged.
> > > 
> > > 
> > > Sure @loadams , i will update. I verified this patch on v1.16.0, as our dev process is on the next release branch, and i did not verified v1.14.0 (the release which is being tested here). There was a lot of bug fixed since then, cannot say for sure what is the root cause.
> > 
> > 
> > Thanks @nelyahu - feel free to tag us when it the new version is released and updated so we can merge this then, thanks!
> 
> @nelyahu - I tried 1.15.1, looks like same errors, so we will wait for 1.16

@loadams Yes.

loadams commented:
@nelyahu - now that we have updated the hpu runner to 1.17, should we move ahead with merging this PR?

loadams commented:
> @nelyahu - now that we have updated the hpu runner to 1.17, should we move ahead with merging this PR?

@nelyahu/ @BacharL - looks like there is an error with this, could you take a look when you have time?

nelyahu commented:
> > @nelyahu - now that we have updated the hpu runner to 1.17, should we move ahead with merging this PR?
> 
> @nelyahu/ @BacharL - looks like there is an error with this, could you take a look when you have time?

Yes, this test is disabled locally in our env and fails for the same reason. I removed it from gaudi2 workflow file. once fixed will be re-added
